Sources/ResilientDecoding/Resilient.swift

```diff
@@ -81,38 +69,21 @@ extension Decoder {
    Since this is the only way to create a `Resilient` with an error, this ensures we are reporting all errors encountered in this manner.
    */
   func resilient<T>(_ fallbackValue: T, error: Error) -> Resilient<T> {
-    resilientDecodingHandled(error)
     #if DEBUG
-      return Resilient(
-        value: fallbackValue,
-        topLevelError: error,
-        errorsAtOffset: [])
+      if error is ArrayDecodingError {
+        /// Resilient arrays are reponsible for reporting element decoding errors themselves.
```

Comment:
Can you help me understand how this  works in practice?

Reply:
`ArrayDecodingError` is a special `Resilient` construct which represents some number of errors that were reported when decoding the elements of the array. We don't want to report both those errors _and_ the collated error separately.